### PR TITLE
Support streaming response body to stream destination and remove dependencies

### LINKED ISSUE #5: Throws error if no option is given
## Issue
Throws error if no options is given. The snippet below throws error 

```js
const { createClient } = require("cmsdata-client");

(async function main() {
const CMSClient = createClient('5fr6-cch3');  // throws if no option is passed

  const dataset = await CMSClient.select().get();

})()


```
## Expected Behavior
Should default to `json` as output type. This issue occurs on line 13 which checks for the the output type.

## Actual Behavior
Throws error when no options is given. 

Screen Shot 2021-06-07 at 4 02 57 PM

## Specifications

  - Version: 1.1.1
  - Platform: macOS Big Sur

### PR DESCRIPTION
**Pull request by hawyar**
I decided to replace `cross-fetch` with `node-fetch` and to only support node environments for now. I think we should worry being isomorphic in the future. Maybe as a  __good first issue__?

Summary of commits: 
1. Remove unused dependencies
2. Fix #5 
3. Add support to stream response body by piping body -> stream destination
4. Merge into master

### CODE REVIEW

**Comment by hawyar** [comment]
Quick recap: 
- Initially added support for browser, then @shanejearley and I decided to move forward without it. We only support node env for now
- Added testing on push to master and pull request
- Merged into master

**Review by shanejearley — APPROVED** [review]
Great. We'll merge this PR and then continue working to resolve the browser compatibility aspects of #4.

File: index.test.js [hunk 1]
```diff
@@ -1,5 +1,7 @@
 const { test } = require('tap');
```
Comment (ccali11) [inline index.test.js:1]: @hawyar - I haven't used tap before. Looks cool. Tell me about why you like it!

File: index.test.js [hunk 2]
```diff
@@ -20,6 +22,22 @@ test('Query all records without metadata', async (t) => {
...
+  t.ok(dataset.data, 'Returns all records with metadata');
+});
+
+test('Query all records with csv output', async (t) => {
+  const CMSClient = createClient('5fr6-cch3', {
+    output: 'csv',
+  });
+
+  const dataset = await CMSClient.select().get();
+  t.ok(dataset.data, 'Returns all records in csv format');
```
Comment (ccali11) [inline index.test.js:38]: According to [docs](https://node-tap.org/docs/api/asserts/), `t.ok` is just verifying that the value passed in is truthy. Is this actually testing if the function returns "all records in a csv format?" All good if you're just scaffolding some tests to add more complexity later. I'm cool with that.

**Review by ccali11 — APPROVED** [review]
Nice job, @hawyar! All looks reasonable to me. Let me know your thoughts on tap and then the specific question I have regarding meaningfulness of (at least) one of the tests. Both questions are non-blocking because code is still valid; good to merge.

File: index.test.js (hunk 2 above)
Comment (hawyar) [reply index.test.js:38]: I am just scaffolding them right now. I am using `t.ok` to give me slight confidence between the changes I make. For the next PR, I am replacing `t.ok` with `t.equal` and also checking for the correct output type, as you mentioned. You are right; these tests are not correct and a bit misleading. I will fix it asap!

File: index.test.js (hunk 1 above)
Comment (hawyar) [reply index.test.js:1]: Small API and easy to use. I don't like connecting three tools for some non-complex testing. Tap gets the job done without a headache.

**Pull request merged**